src/vdom/component.js

```diff
@@ -102,70 +102,199 @@ export function renderComponent(component, opts, mountAll, isChild) {
 			context = extend(extend({}, context), component.getChildContext());
 		}
 
-		let childComponent = rendered && rendered.nodeName,
-			toUnmount, base;
+		if (Array.isArray(rendered)) {
+			component.parent = parent;
+			// We need to figure out how to compare these against the parent's base
+			let toUnmount = false;
+			let base = [];
+			rendered.forEach((child, i) => {
+				let renderedChild = child;
+				let childComponent = renderedChild && renderedChild.nodeName, toUnmount;
+
+				if (typeof childComponent === 'function') {
+					// set up high order component link
+
+					let childProps = getNodeProps(renderedChild);
+					inst = initialChildComponent;
+
+					if (
+						inst &&
+						inst.constructor === childComponent &&
+						childProps.key == inst.__key
+					) {
+						setComponentProps(inst, childProps, SYNC_RENDER, context, false);
+					} else {
+						toUnmount = inst;
+
+						component._component = inst = createComponent(
```

Comment:
I think that _component may have to become an array for this to work. createComponent could accept an array and always return an array.